spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/quartz/TracingJobListener.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.quartz;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import brave.Span;
+import brave.Tracer;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobListener;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * {@link org.quartz.JobListener JobListener} that will wrap a span around quartz jobs
+ * when they start and finish.
+ *
+ * @author Branden Cash
+ */
+@Component
+public class TracingJobListener implements JobListener {
+
+	private final Map<String, Span> jobSpans = new HashMap<>();
```

Comment:
If keeping the map you will need to use the Concurrent variant to ensure thread safety

Reply:
Good suggestion, but I like putting it in the `JobExecutionContext` way better